test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/RuntimeTagHelperDescriptorFactoryTest.cs

```diff
@@ -0,0 +1,158 @@
+                                AssemblyName = AssemblyName,
+                                DesignTimeDescriptor = new TagHelperDesignTimeDescriptor
+                                {
+                                    OutputElementHint = "strong"
+                                }
+                            }
+                        }
+                    },
+                    {
+                        typeof(MulitpleDescriptorTagHelperWithOutputElementHint),
+                        new[]
+                        {
+                            new TagHelperDescriptor
+                            {
+                                TagName = "a",
+                                TypeName = typeof(MulitpleDescriptorTagHelperWithOutputElementHint).FullName,
+                                AssemblyName = AssemblyName,
+                                DesignTimeDescriptor = new TagHelperDesignTimeDescriptor
+                                {
+                                    OutputElementHint = "div"
+                                }
+                            },
+                            new TagHelperDescriptor
+                            {
+                                TagName = "p",
+                                TypeName = typeof(MulitpleDescriptorTagHelperWithOutputElementHint).FullName,
+                                AssemblyName = AssemblyName,
+                                DesignTimeDescriptor = new TagHelperDesignTimeDescriptor
+                                {
+                                    OutputElementHint = "div"
+                                }
+                            }
+                        }
+                    }
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(OutputElementHintData))]
+        public void CreateDescriptors_CreatesDesignTimeDescriptorsWithOutputElementHint(
+            Type tagHelperType,
+            TagHelperDescriptor[] expectedDescriptors)
+        {
+            // Arrange
+            var errorSink = new ErrorSink();
+
+            // Act
+            var descriptors = TagHelperDescriptorFactory.CreateDescriptors(
+                AssemblyName,
+                GetTypeInfo(tagHelperType),
+                designTime: true,
+                errorSink: errorSink);
+
+            // Assert
+            Assert.Empty(errorSink.Errors);
+
+            // We don't care about order. Mono returns reflected attributes differently so we need to ensure order
+            // doesn't matter by sorting.
+            descriptors = descriptors.OrderBy(descriptor => descriptor.TagName);
+
+            Assert.Equal(expectedDescriptors, descriptors, CaseSensitiveTagHelperDescriptorComparer.Default);
+        }
+#endif
+
+        [Theory]
+        [MemberData(nameof(TagHelperWithPrefixData))]
+        public void CreateDescriptors_WithPrefixes_ReturnsExpectedAttributeDescriptors(
```

Comment:
Tests exists with some additional code to make it succeed in CoreCLR - https://github.com/aspnet/Razor/pull/500/files#diff-167cc0b25a0b60456ca01e589f30fd1fR62